py4cast/datasets/base.py

```diff
@@ -714,6 +718,112 @@ def dataset_extra_statics(self) -> List[NamedTensor]:
         """
         return []
 
+    def day_of_years(self, date: dt.datetime, terms: List[float]) -> np.array:
+        """
+        Compute the day of the year for the specified terms from the date.
+        First january is 1, not 0.
+        """
+        days = []
+
+        for term in terms:
+            date_tmp = date + dt.timedelta(hours=float(term))
+            starting_year = dt.datetime(date_tmp.year, 1, 1)
+            days.append((date_tmp - starting_year).days + 1)
+
+        return np.asarray(days)
+
+    def hours_of_day(self, date: dt.datetime, terms: List[float]) -> np.array:
+        """
+        Compute the hour of the day for the specified terms from the date.
+        """
+        hours = []
+        for term in terms:
+            date_tmp = date + dt.timedelta(hours=float(term))
+            hours.append(date_tmp.hour + date_tmp.minute / 60)
+        return np.asarray(hours)
+
+    def seconds_from_start_of_year(
+        self, date: dt.datetime, terms: List[float]
+    ) -> np.array:
+        """
+        Compute how many seconds have elapsed since the beginning of the year for the specified terms.
+        """
+        start_of_year = dt.datetime(date.year, 1, 1)
+        return np.asarray(
+            [
+                (date + dt.timedelta(hours=float(term)) - start_of_year).total_seconds()
+                for term in terms
+            ]
+        )
+
+    def get_year_hour_forcing(
+        self, date: dt.datetime, terms: List[float]
+    ) -> torch.Tensor:
+        """
+        Get the forcing term dependent of the date and terms.
+        """
+        hours_of_day = self.hours_of_day(date, terms)
+        seconds_from_start_of_year = self.seconds_from_start_of_year(date, terms)
+
+        hour_angle = (torch.Tensor(hours_of_day) / 12) * torch.pi  # (sample_len,)
+        year_angle = (
+            (torch.Tensor(seconds_from_start_of_year) / SECONDS_IN_YEAR) * 2 * torch.pi
+        )  # (sample_len,)
```

Comment:
Can you check the `get_year_hour_forcing` method in Titan ? I've made a modification to compute the number of days in a year, including leap years. So we can stop using the variable `SECONDS_IN_YEAR` which doesn't account for leap years.